uitools/import/Esri/ArcGISRuntime/Toolkit/Controller/BookmarksViewController.qml

```diff
@@ -0,0 +1,107 @@
+/*******************************************************************************
+ *  Copyright 2012-2022 Esri
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ ******************************************************************************/
+import QtQml 2.15
+import Esri.ArcGISRuntime 100.15
+import QtQml.Models 2.15
+
+/*!
+   \qmltype BookmarkController
+   \inqmlmodule Esri.ArcGISRuntime.Toolkit
+   \since Esri.ArcGISRuntime 100.15
+   \ingroup ArcGISQtToolkitUiQmlControllers
+   \brief The controller part of a BookmarksView. This class handles the
+   management of the Bookmark objects.
+ */
+
+QtObject {
+    id: controller
+    /*!
+      \qmlproperty GeoView geoView
+      \brief The geoview of the controller.
+     */
+    property GeoView geoView: null;
+
+    /*!
+      \qmlproperty ListModel bookmarks
+      \brief The bookmarks of Bookmark list objects.
+
+      Internally, the list is a QML ListModel containing elements of type ListElement.
+      Each ListElement has a single property `modelData`, which maps to a Bookmark.
+     */
+    readonly property alias bookmarks: internal.bookmarks
+
+    /*!
+       \brief Method that takes a Bookmark \a modelData and sets this controller's
+       bookmark to the map contained within. This method is for QML/C++ layer
+       compatibility.
+     */
```

Comment:
❌  Word salad. This is not for QML/C++ layer compatibility either. We take a `bookmark` and we zoom to that bookmark in the `GeoView`.